src/app/subscribe-events.ts

```diff
@@ -62,17 +63,27 @@ export function subscribeToComponentEvents(wrapperCtx: any) {
 
             let highlightQuery: any = undefined;
 
+            const proteinId = wrapperCtx.proteinId;
+            const start = parseInt(e.detail.start);
+            const end = parseInt(e.detail.end);
+
             // Create query object from event data
-            if(e.detail.start && e.detail.end){
-                highlightQuery = {
-                    start_residue_number: parseInt(e.detail.start),
-                    end_residue_number: parseInt(e.detail.end)
-                };
+            if (start && end) {
+                highlightQuery = proteinId
+                    ? {
+                          uniprot_accession: proteinId,
+                          start_uniprot_residue_number: start,
+                          end_uniprot_residue_number: end,
+                      }
+                    : {
+                          start_auth_residue_number: start,
+                          end_auth_residue_number: end,
+                      };
```

Comment:
Not our typical functional style, but of course, let's keep it to minimize merging problems.

Reply:
Yep, for me I would change it to functional style, but I didn't do it for that..

Reply:
Hmm, but actually as TS is now complaining about the type and we have the repetition on the function also, even if I would desire to minimize the merging problems, probably if there are changes from upstream I would like to review dose changes on these specific functions. Do you mind if I change this functions into functional style?

Reply:
I did a quick review, and only 3 checks for undefined were needed, in the old code block. You can add the check, but otherwise I'd keep upstream code imperative as it is, if they ever change this block of code, the merge would be difficult. But for our own code there is no problem, we can abstract to function, whatever we want.